### ISSUE TITLE
PageDown / Up stops woring in gallery, chat list, contact list (`reacti-window`)

### ISSUE THREAD

**Issue by WofWca** [deltachat/deltachat-desktop#4302]
See https://github.com/deltachat/deltachat-desktop/pull/4269#issuecomment-2452119850

As I said in https://github.com/deltachat/deltachat-desktop/pull/4269#issuecomment-2452195277, maybe a solution would be to add explicit PageUp / PageDown shortcuts in such places.
But maybe arrow keys are enough anyway, perhaps it's not that important.

- Operating System (Linux/Mac/Windows/iOS/Android):
- Delta Chat Version: https://github.com/deltachat/deltachat-desktop/commit/05ee7d59c463c6574549524302f75a01104c657a
- Expected behavior:
    Page Up / PageDown always works
- Actual behavior:
    Page Up / PageDown doesn't work after a bit of scrolling
- Steps to reproduce the problem:
    1. Open gallery (you must have a lot of items there)
    2. Focus an element in gallery using keyboard
    3. Press Page Down 2 times to scroll down
- Screenshots:
    
    https://github.com/user-attachments/assets/9f8e72f9-d669-442f-bd01-a6d650390f19

- Logs:

This is happening because the element gets unrendered (thanks to `react-window`), and loses focus.

**Comment by WofWca** [comment]
#4438 could be somewhat beneficial IMO. Insted of scrolling to the very top and thus losing focus, it will now scroll to the first / last rendered element. So you can hold Home / End to go to the start end, though yes, you'd need to hold it and not just press once.

**Comment by r10s** [comment]
> maybe a solution would be to add explicit PageUp / PageDown shortcuts in such places

the user expectation is clearly that the _normal_ navigation controls work in the upper, focused window - so cursor, pos1/end, page keys, etc.

that paging modifies the _underlying_  window is unexpected.

trying to fix that by different "PageUp / PageDown shortcuts in such places" would just increase confusion.

as a general note: we should also stop adding these special shortcuts for now, this gets too nerdy, we're not emacs or vim. instead, we should concentrate on making the normal navigation, that the average user knows about, working. eg. also arrow keys are not working as expected in the gallery - eg. down goes left etc.

**Comment by WofWca** [comment]
What do you mean by "underlying" and "upper" windows in this case?

**Comment by r10s** [comment]
that keyboard shortcuts change the barely visible message list, where the focus is clearly the gallery, covering most of the screen.

**Comment by WofWca** [comment]
This bug report doesn't have to do with anything happening outside of the gallery dialog, specifically the message list or the chat list.

**Comment by WofWca** [comment]
> This bug report

Rather "the reproduction example".

**Comment by WofWca** [comment]
I'm contemplating upgrading to `react-window` V2 and re-opening this closed MR https://github.com/bvaughn/react-window/pull/549 as a way forward.

**Comment by WofWca** [comment]
I recorded my attempt at adding PageUp / PageDown handling to the roving tabindex hook here https://github.com/deltachat/deltachat-desktop/pull/6541. The problem with it is that I think it really needs to be a proper implementation, especially for the grid widget, as opposed to simply having a fixed jump size regardless of how big the "page" actually is. See the TODO comment there.

About the idea of `react-window`: the library needs some work. The maintainers of it don't seem to be keeping in mind that list items could be interactive / focusable / stateful. I edited the issue with links to some relevant MRs and issues.

Given the recent "minimize the amount of issues and MRs" project goal, I think it's going to be better for Delta Chat as a product to close the issue now and prioritize something else. The bug is sure annoying, but is not too bad, and properly fixing it would take a lot.
